benchmarks/common.py

```diff
@@ -1530,145 +1395,73 @@ def main(runner, original_dir=None):
-        experiment = speedup_experiment
-        backend_str = "nvfuser" if args.nvfuser else "nnc"
-        output_filename = f"accuracy_aot_{backend_str}.csv"
-    elif args.accuracy_aot_ts_mincut:
-        optimize_ctx = torchdynamo.optimize("aot_nvfuser", nopython=args.nopython)
-        # accuracy_ctx = torchdynamo.optimize(
-        #     "aot_nvfuser_nodecomps", nopython=args.nopython
-        # )
-        experiment = speedup_experiment
-        assert args.nvfuser, "TODO - Add another aot string for mem fusion with NNC"
-        backend_str = "nvfuser" if args.nvfuser else "nnc"
-        output_filename = f"accuracy_aot_{backend_str}_mincut.csv"
-    elif args.prims_nvfuser:
-        optimize_ctx = torchdynamo.optimize("prims_nvfuser", nopython=args.nopython)
-        experiment = speedup_experiment
-        backend_str = "prims_nvfuser"
-        output_filename = f"accuracy_aot_{backend_str}.csv"
-    elif args.print_fx:
-        optimize_ctx = torchdynamo.optimize(
-            print_fx,
-            nopython=args.nopython,
-        )
-    elif args.print_aten_ops:
-        optimize_ctx = torchdynamo.optimize(
-            print_aten_ops,
-            nopython=args.nopython,
-        )
-    elif args.accuracy_ts:
-        optimize_ctx = torchdynamo.optimize(fixed_strategy1, nopython=args.nopython)
-        experiment = speedup_experiment
-        backend_str = "nvfuser" if args.nvfuser else "nnc"
-        output_filename = f"accuracy_{backend_str}.csv"
-    elif args.nothing:
-        pass
-    elif args.nops:
-        optimize_ctx = torchdynamo.eval_frame._optimize_catch_errors(
-            torchdynamo.testing.debug_insert_nops, nopython=args.nopython
-        )
-    elif args.backend:
-        optimize_ctx = torchdynamo.optimize(args.backend, nopython=args.nopython)
-        experiment = speedup_experiment
-        output_filename = f"speedup_{args.backend}.csv"
-    elif args.log_conv_args:
-        optimize_ctx = torchdynamo.optimize(conv_args_analysis, nopython=args.nopython)
-        output_filename = "log_conv_args.csv"
-    elif args.recompile_profiler:
-        output_filename = "recompile_profiler_log.csv"
-        experiment = recompile_profiler_experiment
-    else:
-        optimize_ctx = torchdynamo.optimize(fx_insert_profiling, nopython=args.nopython)
-        experiment = coverage_experiment
-        output_filename = "coverage.csv"
+
+    # these names are based on the historical command line flag values
+    backends = {
+        "overhead": dummy_fx_compile,
+        "online_autotune": online_autotuner,
+        "offline_autotune": offline_autotuner,
+        "python_key": python_key,
+        "ltc": backends.ltc_reuse_graph,
+        "ltc_trivial": backends.ltc_trivial,
+        "fixed1": fixed_strategy1,
+        "fixed2": fixed_strategy2,
+        "ts": backends.ts,
+        "fx2trt": backends.fx2trt_compiler,
+        "fx2trt_fp16": backends.fx2trt_compiler_fp16,
```

Comment:
should control this with the --float16 flag not a separate target.

Reply:
That's kind of inelegant though, because it breaks the invariant that the backend argument maps to a particular torchdynamo.optimize argument; instead, that argument might wobble depending on other flags. Seems better to not do that, especially since this isn't a cowpath backend?